### PR TITLE
Add interfaces to connect Local/RemoteTrack and RTPSender/Receiver

### PR DESCRIPTION
**Pull request by at-wat**
### Description
Add abstract interfaces LocalTrack, RemoteTrack and adapter to connect Track and RTPSender/Receiver.
Add some example pseudo code to connect LocalRTPTrack and PassthroughRTPSender.

### Reference issue
https://github.com/pion/webrtc-v3-design/issues/23#issue-694723158

### CODE REVIEW

File: rtpengine/io.go [hunk 1]
```diff
@@ -16,3 +16,8 @@ type Writer interface {
 	WriteRTP(*rtp.Packet) error
 	ReadRTCP() (rtcp.Packet, error)
 }
+
+// Copy rtp.Packet from Reader to Writer and rtcp.Packet from Writer to Reader.
+func Copy(Writer, Reader) error {
```
Comment (at-wat) [inline rtpengine/io.go]: It should be cancelable on ReplaceTrack().

File: webrtc/track.go [hunk 2]
```diff
@@ -10,10 +10,38 @@ type TrackBase interface {
...
+	//     }
+	//     ptSender, ok := sender.(PassthroughRTPSender)
+	//     if !ok {
+	//       return ErrIncompatible
+	//     }
+	//     // RTP packets written via LocalRTPTrack.WriteRTP() will be
+	//     // read by LocalRTPTrack.pipeReader.ReadRTP().
+	//     return rtpengine.Copy(ptSender, rtpTrack.pipeReader)
+	//   }
+	Bind(LocalTrack, RTPSender) error
```
Comment (at-wat) [inline webrtc/track.go]: It should have Unbind for ReplaceTrack.

**Comment by at-wat** [comment]
@Sean-Der @lherman-cs please take a look

**Comment by Sean-Der** [comment]
@at-wat nice this looks really good to me! I will try to finish the examples this week.

You should push without a review! I need to put more effort into finishing this repo (so we can release `v3`) the `v2` branch is getting quite diverged :(

**Pull request merged**